packages/backend/src/peripherals/arbitrum/multicall/ArbitrumMulticall.ts

```diff
@@ -0,0 +1,94 @@
+import { EthereumAddress } from '@l2beat/shared-pure'
+
+import { EthereumClient } from '../../ethereum/EthereumClient'
+import {
+  MulticallEncoder,
+  MulticallRequest,
+  MulticallResponse,
+} from './interfaces'
+import { arbitrumMulticallEncoder } from './MulticallEncoder'
+
+/**
+ * @see https://arbiscan.io/address/0x842eC2c7D803033Edf55E478F461FC547Bc54EB2
+ */
+export const ARBITRUM_MULTICALL_BATCH_SIZE = 150
+export const ARBITRUM_MULTICALL_BLOCK = 821923
+export const ARBITRUM_MULTICALL_ADDRESS = EthereumAddress(
+  '0x842eC2c7D803033Edf55E478F461FC547Bc54EB2',
+)
+
+export class ArbitrumMulticallClient {
```

Comment:
Two things: 
 - We should be able to refactor Multicall in the near future to be fully reusable, or at least to some degree. Now ArbitrumMulitcall is altered and slightly polished. Refactoring this may take some more thinking since Ethereum Multicall consists of two separate versions of underlying call contract. However, it is doable
 - Once we agree what to do with Multicalls it would be great to prepare some configuration management for those

Reply:
Let's prioritize this refactor after we have the requested USDC balance ready. I created an issue for that https://linear.app/l2beat/issue/L2B-1788/refactor-arbitrummulticall-and-ethereummulticall